src/main/java/org/fogbowcloud/manager/core/threads/OpenProcessor.java

```diff
@@ -0,0 +1,177 @@
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+
+public class OpenProcessor implements Runnable {
+
+	private InstanceProvider localInstanceProvider;
+	private InstanceProvider remoteInstanceProvider;
+
+	private ChainedList openOrdersList;
+	private ChainedList pendingOrdersList;
+	private ChainedList failedOrdersList;
+	private ChainedList spawningOrdersList;
+
+	private String localMemberId;
+
+	/**
+	 * Attribute that represents the thread sleep time when there is no Orders
+	 * to be processed.
+	 */
+	private Long sleepTime;
+
+	private static final Logger LOGGER = Logger.getLogger(OpenProcessor.class);
+
+	public OpenProcessor(InstanceProvider localInstanceProvider, InstanceProvider remoteInstanceProvider,
+			String localMemberId, Properties properties) {
+		this.localInstanceProvider = localInstanceProvider;
+		this.remoteInstanceProvider = remoteInstanceProvider;
+		this.localMemberId = localMemberId;
+
+		// TODO: ChainedLists instanciation by Singleton pattern.
+
+		String schedulerPeriodStr = properties.getProperty(ConfigurationConstants.OPEN_ORDERS_SLEEP_TIME_KEY,
+				DefaultConfigurationConstants.OPEN_ORDERS_SLEEP_TIME);
+		this.sleepTime = Long.valueOf(schedulerPeriodStr);
+	}
+
+	@Override
+	public void run() {
+		while (true) {
+			try {
+				Order order = this.openOrdersList.getNext();
+				if (order != null) {
+					this.processOpenOrder(order);
+				} else {
+					LOGGER.info("There is no Open Order to be processed, sleeping OpenProcessor Thread...");
+					Thread.sleep(this.sleepTime);
+				}
+			} catch (Throwable e) {
+				LOGGER.error("Error while trying to Process an Open Order with the OpenProcessor Thread", e);
+			}
+		}
+	}
+
+	/**
+	 * Method that try to get an Instance for an Open Order. This method can
+	 * generate a race condition. For example: a user can delete a Open Order
+	 * while this method is trying to get an Instance for this Order.
+	 */
+	protected void processOpenOrder(Order order) {
+		synchronized (order) {
+			OrderState orderState = order.getOrderState();
+
+			if (orderState.equals(OrderState.OPEN)) {
+				LOGGER.info("Trying to get an Instance for Order [" + order.getId() + "]");
+
+				try {
+					InstanceProvider instanceProvider = this.getInstanceProviderForOrder(order);
+
+					LOGGER.info("Processing Order [" + order.getId() + "]");
+					order.handleOpenOrder();
+					OrderInstance orderInstance = instanceProvider.requestInstance(order);
+					order.setOrderInstance(orderInstance);
+
+					LOGGER.info("Removing Order [" + order.getId() + "] from Open Orders List");
+					this.openOrdersList.removeItem(order);
+
+					LOGGER.info("Updating Order State after processing [" + order.getId() + "]");
+					this.updateOrderStateAfterProcessing(order);
+
+				} catch (Exception e) {
```

Comment:
This exception will never be thrown, right? Could you change responsible method signature to do that? Plus, after the Exception being thrown, we need to remove order from openOrderList.

Reply:
@GustavoYamaguchi no, if the OrderInstance returned by the Instance be null or your Id attribute be an empty string, it will be thrown.

Reply:
Exactly, in the interface we need to add a 'throws Exception', to requestInstance() method.

Reply:
Done, [here](https://github.com/fogbow/fogbow-manager-core/pull/13/commits/6eaf68cea65625b38e7ad8f611eca9b3beff6e40).